app/models/accounting/quickbooks/journal_entry_extractor.rb

```diff
@@ -60,6 +59,10 @@ def account
           txn.line_items.find(&:debit?).account
         when 'disbursement'
           txn.line_items.find(&:credit?).account
+        when 'other'
+          # use find or create to avoid repeat entries
+          ::Accounting::ProblemLoanTransaction.find_or_create_by(loan: loan, accounting_transaction: txn, error_message: :journal_entry_type_other_account_not_set)
+          nil
```

Comment:
I was aiming for the lightest touch on the code possible for this. I thought about using a uniqueness constraint in the db and/or validation in the model, and catching an error, but that seemed not quite right. I also thought about adding a wrapper method in the ProblemLoanTransaction model or a helper PORO. Suggestions?  Also, does this mean I need to index on error message in the db?

Reply:
I don't think we care much about duplicates here. Because of the grouping behavior in the view, it doesn't much matter if the error shows for the same loan and txn more than once, it just takes slightly more vertical space, and as with the other thing, this table will get wiped and reset relatively frequently, especially since resetting the QB data is the recommended way to fix errors that get corrected on the QB side. I also am not anticipating this table getting large enough for us to care deeply about the performance. Indexing and a uniqueness constraint wouldn't hurt, but I just don't think we care that much. This table is a journal/log, not an integral part of the business rules of the app, it's okay if it's sort of a dump site, the data integrity is not our biggest concern.